Formula/s-nail.rb

```diff
@@ -0,0 +1,75 @@
+class SNail < Formula
+  desc "Fork of Heirloom mailx."
+  homepage "https://www.sdaoden.eu/code.html"
+  url "https://www.sdaoden.eu/downloads/s-nail-14.9.3.tar.gz"
+  sha256 "9048abe94c8b732ddefcd70b9f6052da16977de76b07598790e4c763e97f911d"
+
+  # Define options
+  option "with-maximal", "Anything on, including experimental features."
+  option "with-netsend", "Sending messages directly to the mail provider via the SMTP protocol, instead of requiring a local mail-transfer-agent (MTA) who does."
+  option "with-medium", "Possibly what people want who need nothing but a MIME-capable mailx(1)."
+  option "with-minimal", "This is the most plain mailx(1)-alike mode, but with MIME support."
+  option "with-nulli", "Anything that can be turned off is off but iconv(3)."
+  option "with-null", "Anything that can be turned off is off."
+
+  # Formula depends on OpenSSL per default and in MAXIMAL and NETSEND
+  depends_on "openssl" if build.with?("maximal") || build.with?("netsend") || (build.without?("medium") && build.without?("minimal") && build.without?("nulli") && build.without?("null"))
+  # Formula depends on IDNA per default and in MAXIMAL, NETSEND and MEDIUM
+  depends_on "libidn" if build.with?("maximal") || build.with?("netsend") || build.with?("medium") || (build.without?("minimal") && build.without?("nulli") && build.without?("null"))
+
+  def install
+    args = %W[
+      VAL_PREFIX=#{prefix}
+      VAL_PRIVSEP_USER=#{ENV["USER"]}
```

Comment:
This is all related to Dovecot's `dotlock` (see https://wiki2.dovecot.org/MailboxFormat/Maildir). In order for `dotlocking` to be working the helper program `s-nail-privsep` needs rw-privileges on the the `/var/mail`-directory. So there are two options:

- compile without dotlocking but then there might be issues with dovecot, which is the standard mail server on macOS.

- compile with dotlocking. Here I have the problem that `s-nail-privsep` needs to be owned by root and use SETUID, otherwise it is not able to read/write in the `/var/mail/$user` dirs. I passed the optional username as a workaround so that I don't get an error during compile time about missing permissions.

If you insist on not using chown and SETUID than I'll exclude dotlocking. Otherwise I will check with the developer on how we can fix the username issue. Let me know.